diff --git a/CMakeLists.txt b/CMakeLists.txt
index 21aa319..87b8a38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,7 @@ option(ENABLE_WARNINGS "Enable warnings" ON)
 option(ENABLE_NATIVE "Enable native CPU build tuning" OFF)
 option(ENABLE_OPENMP "Enable OpenMP" ON)
 option(ENABLE_KOKKOS "Enable Kokkos" ON)
-option(ENABLE_BLAS "Enable BLAS" OFF)
+option(ENABLE_BLAS "Enable BLAS" ON)
 option(ENABLE_PYTHON "Enable compilation of the Python module" ON)
 option(ENABLE_GATE_DISPATCHER "Enable gate kernel dispatching on AVX/AVX2/AVX512" ON)
 
